### PR TITLE
TD-2968 Remove signoff requests

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-2968

### Description
I added a query to remove the signoff request made when the is change in the confirm self-assessment

### Screenshots
_Attach screenshots on mobile, tablet and desktop._

-----
### Developer checks
image

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs [hunk 1]
```diff
@@ -433,6 +433,7 @@ public IActionResult SelfAssessmentOverview(int selfAssessmentId, string vocabul
 
             var optionalCompetencies = selfAssessmentService.GetCandidateAssessmentOptionalCompetencies(selfAssessmentId, delegateUserId);
             selfAssessmentService.UpdateLastAccessed(assessment.Id, delegateUserId);
+            selfAssessmentService.RemoveSignoffRequests(assessment.Id, delegateUserId);
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs:436]: this is the landing page when you launch any self-assessment. The changes are to be done when you click send a request for confirming proficiencies (Request proficiency confirmation)

**Review by rshrirohit — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs [hunk 2]
```diff
@@ -142,6 +142,17 @@ public void UpdateLastAccessed(int selfAssessmentId, int delegateUserId)
             }
         }
 
+        public void RemoveSignoffRequests(int selfAssessmentId, int delegateUserId)
+        {
+            var numberOfAffectedRows = connection.Execute(
+                @"DELETE A FROM CandidateAssessmentSupervisorVerifications A
+                  INNER JOIN CandidateAssessments B
```
[the remark this answers is not in the document]
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs]: Having taken time to look at the join after our call, I could see that joining the two table by ID field is wrong cause both ID field have primary key in each table

File: DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs (hunk 1 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs]: Done

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs
Comment (sarathlal-sarangadharan) [inline DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:1]: @sherif-olaboye Please check below where clause, 
 WHERE ((ca.DelegateUserID = 1) AND (ca.SelfAssessmentID = 4) AND (sasr.SelfAssessmentReview = 1) AND CG.ID =@competencyGroupId AND casv.SignedOff =0)
                        OR ((ca.DelegateUserID = 1) AND (ca.SelfAssessmentID = 4) AND (sasr.SelfAssessmentReview IS NULL) AND CG.ID =@competencyGroupId AND casv.SignedOff =0)

Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:1]: Thanks for the observation I have added the bracket to the where clause

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs [hunk 3]
```diff
@@ -142,6 +145,39 @@ public void UpdateLastAccessed(int selfAssessmentId, int delegateUserId)
...
+                        ON cas.SupervisorDelegateId = sd.ID
+                    INNER JOIN AdminUsers AS au
+                        ON sd.SupervisorAdminID = au.AdminID
+                    LEFT OUTER JOIN SelfAssessmentSupervisorRoles AS sasr
+                        ON cas.SelfAssessmentSupervisorRoleID = sasr.ID
+						  INNER JOIN SelfAssessmentStructure AS SAS 
+						  ON CA.SelfAssessmentID = SAS.SelfAssessmentID
+                    INNER JOIN CompetencyGroups AS CG 
+					ON SAS.CompetencyGroupID = CG.ID AND SAS.SelfAssessmentID =@selfAssessmentId
+                    WHERE ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview = 1) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0)
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:167]: Please use the braces correctly. See the sample provided by Sarathlal - we need to segregate the bunch of ANDs from the OR. Like 
( (xyz) AND (xyz) AND (xyz) ) OR ( (xyz) AND (xyz) AND (xyz) )

File: DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs [hunk 4]
```diff
@@ -1355,6 +1356,7 @@ public IActionResult WithdrawSupervisorVerificationRequest(int selfAssessmentId,
         [HttpPost]
         public IActionResult SubmitVerification()
         {
+            var delegateUserId = User.GetUserIdKnownNotNull();
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Controllers/LearningPortalController/SelfAssessment.cs:1359]: remove this line since this is an unnecessary change. In this method, you have introduced some line spaces as well which are not required.

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please make the changes requested.

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs (hunk 3 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs]: Done

**Review by rshrirohit — COMMENTED** [review]
Are the libraries needed?
